src/effects/VST3/VST3UIValidator.cpp

```diff
@@ -181,7 +181,7 @@ void VST3UIValidator::OnClose()
 
 bool VST3UIValidator::UpdateUI()
 {
-   mAccess.ModifySettings([&](EffectSettings& settings) { mWrapper.FetchSettings(settings); });
+   mWrapper.FetchSettings(mAccess.Get());
```

Comment:
The problem is that some plugins use `IComponent::performEdit` to deliver the changes to the host (that should be followed by `IAudioProcessor::process`, for that I need `ModifySettings`) and do not update internal processing state during `IAudioProcessor::setState`